sdk/synapse/Azure.Analytics.Synapse.Spark/samples/Sample2_ExecuteSparkStatement.md

```diff
@@ -0,0 +1,200 @@
+```
+
+## Create Spark session
+
+All `SparkStatement` are created within the context of a `SparkSession`, so first create a spark session passing in details on the requested host.
+
+```C# Snippet:CreateSparkSession
+SparkSessionOptions request = new SparkSessionOptions(name: $"session-{Guid.NewGuid()}")
+{
+    DriverMemory = "28g",
+    DriverCores = 4,
+    ExecutorMemory = "28g",
+    ExecutorCores = 4,
+    ExecutorCount = 2
+};
+
+SparkSession sessionCreated = client.CreateSparkSession(request);
+
+// Waiting session creation completion
+sessionCreated = PollSparkSession(client, sessionCreated);
+```
+
+## Retrieve a Spark Session
+
+To retrieve an existing session call `GetSparkSession`, passing in the session ID.
+
+```C# Snippet:GetSparkSession
+SparkSession session = client.GetSparkSession(sessionCreated.Id);
+Debug.WriteLine($"Session is returned with name {session.Name} and state {session.State}");
+```
+
+## Creating Spark statements
+
+To create statements within a session call `CreateSparkStatement`, passing in both the statements details in a `SparkStatementOptions` along with the ID of the session.
+
+To wait for the statement's completion a `PollSparkStatement` call is required. This support code will be detailed below, and it hoped to be temporary. 
+
+```C# Snippet:CreateSparkStatement
+SparkStatementOptions sparkStatementRequest = new SparkStatementOptions
+{
+    Kind = SparkStatementLanguageType.Spark,
+    Code = @"print(""Hello world\n"")"
+};
+SparkStatement statementCreated = client.CreateSparkStatement(sessionCreated.Id, sparkStatementRequest);
+
+// Wait operation completion
+statementCreated = PollSparkStatement(client, sessionCreated.Id, statementCreated);
+```
+
+## Retrieve a statement
+
+To retrieve an existing statement call `GetSparkStatement`, passing in both the session ID and the ID of the statement.
+
+```C# Snippet:GetSparkStatement
+SparkStatement statement = client.GetSparkStatement(sessionCreated.Id, statementCreated.Id);
+Debug.WriteLine($"Statement is returned with id {statement.Id} and state {statement.State}");
+```
+
+## Cancel a statement
+
+To cancel a submitted statement call `CancelSparkStatement`, passing in both the session ID and the ID of the statement.
+
+```C# Snippet:CancelSparkStatement
+SparkStatementCancellationResult cancellationResult = client.CancelSparkStatement(sessionCreated.Id, statementCreated.Id);
+Debug.WriteLine($"Statement is cancelled with message {cancellationResult.Msg}");
+```
+
+## Cancel a session
+
+To cancel the entire Spark session  call `CancelSparkSession`, passing in the session ID.
+
+```C# Snippet:CancelSparkSession
+Response operation = client.CancelSparkSession(sessionCreated.Id);
+```
+
+## Support Code
+
+Today the following support code is needed to poll the status of submitted Spark statements and sessions.
+
+It is hoped to be temporary.
+
+```C# Snippet:TemporarySparkSupportCode
+ private const string Error = "error";
```

Comment:
This snippet is pretty dense.   It may help to make one big psudo-enum class of string constants to group them instead of having them spread out.   I'm thinking something like:

```csharp
public static class SparkStatus
{
    public const string Dead = "dead";
    public const string Success = "success";
    // ...
    public const string Starting = "starting";
    public const string Waiting = "waiting";
    // ...
}

Reply:
It may also be helpful to define each `IList` grouping of states as members of this class or closer to it, for conceptual locality.   Maybe something like:

```csharp
public static class SparkStatus
{
    public const string Dead = "dead";
    public const string Success = "success";
    // ...
    public const string Starting = "starting";
    public const string Waiting = "waiting";
    // ...

    private static IReadOnlyList<string> FinalStates { get; } =
        new List<string>
         {
             Idle,
             Error,
             Dead,
             Success,
             Killed
         };
} 
```

Reply:
I'm looking into converting this to a LRO right now, and I'm making ok progress. Planning on leaving this for now, and revisiting if the LRO works stalls.

Reply:
There is a lot of work to clean up the test code otherwise (as you and earlier Alex noted).